### PR TITLE
created the contract file for database operations and model schema, I…

### PR DESCRIPTION
**Pull request by jac1013**
## Story / Task
https://github.com/bixlabs/bix-notifications/issues/3 (I think this one https://github.com/bixlabs/bix-notifications/issues/9 is also covered here)

## What this PR does?
* We added a layer for the contract of database and the model schema.

### Before
* It was using the MongoDB implementation directly.

### After
* We use the abstraction instead of a direct implementation.
* Left the same code of MongoDB as an implementation example.

## How to test / reproduce
* We don't have test yet :(.

## TODOs
* Whenever tests framework is added we need to move the example code to be used in the tests.

Thanks.

### CODE REVIEW

**Comment by jac1013** [comment]
Hey @rkmax I think we can consider callback/promises after merging this one, we could also add that support in here but I thought it was out of the scope of this issue, let me know what you think.

File: src/notifications/contracts/database.js
Comment (tatianacmh) [inline src/notifications/contracts/database.js]: Here we could change setAllAsRead to setAllRead,you have setAllUnread ..so..

File: src/notifications/contracts/model-schema.json
Comment (tatianacmh) [inline src/notifications/contracts/model-schema.json:13]: Can we have a sentAt or updatedAt?

File: src/notifications/controller.js
Comment (tatianacmh) [inline src/notifications/controller.js]: This sounds awkward, maybe we could change this text to "Sets all unread notifications as read, basically sets the unread field to false"

**Review by tatianacmh — APPROVED** [review]
Looks good to me! :+1:

File: src/notifications/contracts/model-schema.json
Comment (jac1013) [reply src/notifications/contracts/model-schema.json:13]: Sure @tatianacmh, can we create a separate ticket for that? I just replicated the logic that was already in place, let me know if you want me to add this right away.

File: src/notifications/controller.js
Comment (jac1013) [reply src/notifications/controller.js]: Sure!

File: src/notifications/contracts/model-schema.json
Comment (EddSuarez) [reply src/notifications/contracts/model-schema.json:13]: Agree with @tatianacmh, the `updatedAt` it would be great but I think we should call it "readAt" since the only thing we are going to update is the field `unread`. What do you think @jac1013 ?

Comment (jac1013) [reply src/notifications/contracts/model-schema.json:13]: Can't agree more, if you can @EddSuarez create a ticket for that with the details and the explanation of why you want ``readAt``.

Comment (jac1013) [reply src/notifications/contracts/model-schema.json:13]: https://github.com/bixlabs/bix-notifications/issues/14

Comment (EddSuarez) [reply src/notifications/contracts/model-schema.json:13]: @jac1013 great, I added the current schema in [#14](https://github.com/bixlabs/bix-notifications/issues/14).
Are we ready to merge this PR?

Comment (jac1013) [reply src/notifications/contracts/model-schema.json:13]: No, we are waiting on @rkmax review :1st_place_medal:

**Review by rkmax — APPROVED** [review]
:+1:

**Pull request merged**
